types/index.d.ts

```diff
@@ -533,7 +537,8 @@ declare namespace math {
      * @param value Value to be converted
      * @returns The created number
      */
-    number(value?: string | number | BigNumber | Fraction | boolean | MathCollection | Unit | null): number | MathCollection;
+    number(value?: string | number | BigNumber | Fraction | boolean | Unit | null): MathCollection;
+    number(value?: MathCollection): MathCollection;
```

Comment:
Improve `number()` type so that return type depends if value is a collection or not